pkg/controller/clusterautoscaler/clusterautoscaler_controller.go

```diff
@@ -153,7 +153,7 @@ func (r *Reconciler) Reconcile(request reconcile.Request) (reconcile.Result, err
 		return reconcile.Result{}, nil
 	}
 
-	if err := r.UpdateAutoscaler(ca); err != nil {
+	if err := r.updateAutoscaler(ca, dep); err != nil {
```

Comment:
Both `r.GetAutoscaler` are invoked pretty close to each other so the chance of changing the deployment between the calls is quite small. And, if that happens and `client.Update` complains about `the object has been modified; please apply your changes to the latest version and try again`, the deployment gets reconciled in the next loop.

Reply:
Actually, they are invoked after checks of the err with no other operations in between.  If line 141 returns a deployment, the very next action (other than two boolean statements which resolve false) is to call GetAutoscaler again!

Reply:
I get the reasoning here, but I don't think this is performance critical and we have caching. This makes calling the update harder and more error prone if it were to be called elsewhere. I think the original focus was on the API being friendly.

Reply:
So, if we're going to make `updateAutoscaler()` private and require passing in the deployment to compare, my vote would probably be to just inline it in `Reconcile()` instead.

Reply:
@bison I disagree.  I'm not sure what context we would potentially call updateAutoscaler() if we didn't already have a deployment that we needed to update.

I don't agree with inlining, I like it as-is for now.